### ISSUE TITLE
any classification related to the module?

### ISSUE THREAD

**Issue by sijun-zhou** [JJBOY/BMN-Boundary-Matching-Network#20]
Hi, 
    Very appreciated with your great work!
    I'm just start to learn the paper and your code. It seems that there is no classification for any proposals snips of the video for this network. I am not quite sure am I right?

Thanks in advance!

**Comment by JJBOY** [comment]
Yes, you are right. You need another module to classify the proposals which is not implemented in my code

**Comment by sijun-zhou** [comment]
> Yes, you are right. You need another module to classify the proposals which is not implemented in my code

could you help to provide any reference code on the github that i can have a close study. this would be much help. ^_^

**Comment by pzhren** [comment]
> > Yes, you are right. You need another module to classify the proposals which is not implemented in my code
> 
> could you help to provide any reference code on the github that i can have a close study. this would be much help. 
请问你解决了吗，我也需要后续的proposals的分类代码，非常感谢

**Comment by JJBOY** [comment]
> > > Yes, you are right. You need another module to classify the proposals which is not implemented in my code
> > 
> > 
> > could you help to provide any reference code on the github that i can have a close study. this would be much help.
> > 请问你解决了吗，我也需要后续的proposals的分类代码，非常感谢

We will release the code of our new work [PcmNet: Position-Sensitive Context Modeling Network for Temporal Action Localization](https://arxiv.org/abs/2103.05270) sonn. It will include the code of proposal classification.

**Comment by pzhren** [comment]
Thank you for your reply. Is there any code release plan recently, or do you recommend easy-to-reproducible end-to-end video action localization work?




------------------&nbsp;Original&nbsp;------------------
From:                                                                                                                        "JJBOY/BMN-Boundary-Matching-Network"                                                                                    ***@***.***&gt;;
Date:&nbsp;Thu, Jun 3, 2021 03:28 PM
***@***.***&gt;;
Cc:&nbsp;"Pengzhen ***@***.******@***.***&gt;;
Subject:&nbsp;Re: [JJBOY/BMN-Boundary-Matching-Network] any classification related to the module? (#20)





    
Yes, you are right. You need another module to classify the proposals which is not implemented in my code
  
could you help to provide any reference code on the github that i can have a close study. this would be much help.
 请问你解决了吗，我也需要后续的proposals的分类代码，非常感谢
   
We will release the code of our new work PcmNet: Position-Sensitive Context Modeling Network for Temporal Action Localization. This code will include the code of proposal classification.

**Comment by xxcheng0708** [comment]
> > > > Yes, you are right. You need another module to classify the proposals which is not implemented in my code
> > > 
> > > 
> > > could you help to provide any reference code on the github that i can have a close study. this would be much help.
> > > 请问你解决了吗，我也需要后续的proposals的分类代码，非常感谢
> 
> We will release the code of our new work [PcmNet: Position-Sensitive Context Modeling Network for Temporal Action Localization](https://arxiv.org/abs/2103.05270) sonn. It will include the code of proposal classification.

您好，最近在学习您的PcmNet这篇论文，有个疑问想请教一下，我理解的论文中的PCM模块中先使用了一个Attention模块来提取上下文的语义信息，然后同时又使用了一个Attention模块来提取位置敏感的信息，这样的话一个PCM中包含两个Attention，论文的网络结构中一共使用了6个PCM模块，我按照这个思路自己实现了一下发现参数量和计算量超大，batch_size设置成4都跑不起来，但是我看论文中说在Activity1.3上使用的batch_size是32，想请教一下PCM模块的详细实现方式？或者说最近有代码的开源计划吗？谢谢!

**Comment by JJBOY** [comment]
"PCM模块中先使用了一个Attention模块来提取上下文的语义信息，然后同时又使用了一个Attention模块来提取位置敏感的信息，这样的话一个PCM中包含两个Attention", 语义上下文和位置上下文是在同一个模块中实现的，batch_size=32, 刚好占满两张1080Ti的显存。
我是有开源的计划的，但是paper一直没中，所以迟迟没有开源，paper中之后或者我毕业之后会开源。

**Comment by xxcheng0708** [comment]
> > > > > Yes, you are right. You need another module to classify the proposals which is not implemented in my code
> > > > 
> > > > 
> > > > could you help to provide any reference code on the github that i can have a close study. this would be much help.
> > > > 请问你解决了吗，我也需要后续的proposals的分类代码，非常感谢
> > 
> > 
> > We will release the code of our new work [PcmNet: Position-Sensitive Context Modeling Network for Temporal Action Localization](https://arxiv.org/abs/2103.05270) sonn. It will include the code of proposal classification.
> 
> 您好，最近在学习您的PcmNet这篇论文，有个疑问想请教一下，我理解的论文中的PCM模块中先使用了一个Attention模块来提取上下文的语义信息，然后同时又使用了一个Attention模块来提取位置敏感的信息，这样的话一个PCM中包含两个Attention，论文的网络结构中一共使用了6个PCM模块，我按照这个思路自己实现了一下发现参数量和计算量超大，batch_size设置成4都跑不起来，但是我看论文中说在Activity1.3上使用的batch_size是32，想请教一下PCM模块的详细实现方式？或者说最近有代码的开源计划吗？谢谢!

检查了一下网络模型发现，结合您论文汇总Fig 4.的网络结构描述，我理解的是AvgPool和MaxPool之后得到的是1 x C的输出，然后将1 x C的输出输入到Shared MLP中，然后分别得到两个T x C的输出。所以我在实现时MLP使用了两层的FC层，第一层的输入、输出channel是 [C, T x C]，第二层的输入、输出channel是[T x C，T x C]。当T=100, C=400时，第二层FC的参数量达到的16亿，所以问题出在这里了。

**Comment by JJBOY** [comment]
MLP 我实际上只用了一层fc，而且输入输出的channel都是1xC, 利用广播机制和原始输入相加，来节省计算，。

**Comment by xxcheng0708** [comment]
非常感谢您的解答，谢谢！
